src/applications/model/gaming-streaming-server.cc

```diff
@@ -131,44 +130,44 @@ GamingStreamingServer::Send (int idx)
 {
   NS_LOG_FUNCTION (this);
 
-  uint32_t pktSize = m_trafficStreams[idx].m_packetSizeVariable->GetInteger ();
+  uint32_t pktSize = m_trafficStreams[idx].packetSizeVariable->GetInteger ();
 
   Ptr<Packet> packet = Create<Packet> (pktSize);
   TimestampTag timestamp;
   timestamp.SetTimestamp (Simulator::Now ());
   packet->AddByteTag (timestamp);
   NS_ABORT_IF (packet->GetSize () != pktSize);
 
-  m_socket->Send (packet);
-  m_totSentPackets++;
-  m_totSentBytes += pktSize;
-
   std::stringstream addrString;
-  if (Ipv4Address::IsMatchingType (m_peerAddress))
+  if ((m_socket->Send (packet)) >= 0)
     {
-      addrString << Ipv4Address::ConvertFrom (m_peerAddress);
+      m_totSentPackets++;
+      m_totSentBytes += pktSize;
+
+      if (Ipv4Address::IsMatchingType (m_peerAddress))
+        {
+          addrString << Ipv4Address::ConvertFrom (m_peerAddress);
+        }
+      else if (Ipv6Address::IsMatchingType (m_peerAddress))
+        {
+          addrString << Ipv6Address::ConvertFrom (m_peerAddress);
+        }
+      NS_LOG_INFO("At time " << Simulator::Now ().GetSeconds ()
+                             << "s gaming server sent "
+                             << packet->GetSize () << " bytes to "
+                             << addrString.str ()
+                             << " port " << m_peerPort
+                             << " total sent packets " << m_totSentPackets
+                             << " total Tx " << m_totSentBytes << " bytes");
     }
-  else if (Ipv6Address::IsMatchingType (m_peerAddress))
+  else
     {
-      addrString << Ipv6Address::ConvertFrom (m_peerAddress);
+      NS_LOG_INFO ("Error while sending " << packet->GetSize () << " bytes to "
```

Comment:
What's the content of addrString if the `else `part of this code is executed?

Reply:
you right. the addrString never initialize in else part.